backend/t4cclient/extensions/modelsources/git/routes.py

```diff
@@ -148,12 +148,8 @@ def get_revisions(
         elif ref.startswith("refs/tags/"):
             remote_refs["tags"].append(ref[len("refs/tags/") :])
 
-    if git_model.revision != "HEAD":
-        remote_refs["default"] = git_model.revision
-    elif "master" in remote_refs["branches"]:
-        remote_refs["default"] = "master"
-    else:
-        remote_refs["default"] = "main"
+    remote_refs["default"] = git_model.revision
```

Comment:
I think it would be better to keep the if else statements since we then set in either case a default branch in the frontend. 
Think about when `git_model.revision` is HEAD which would be neither in `remote_refs["tags"]` nor `remote_refs["branches"]`. Then, we would have no branch by default in the frontend. Alternatively, we could think about to resolve the HEAD, but the HEAD could potentially e.g. refs/pull/81/head which would be again not part of `remote_refs["tags"]` or `remote_refs["branches"]` and therefore, no branch would be set by default in the frontend.
Same applies when `git_model.revision` has a typo error which would be then again not included in `remote_refs["tags"]` nor `remote_refs["branches"]`.

Reply:
To take your example: Let's imagine repository where the default branch is `staging`, which is used as `HEAD`. 
If one specifies `HEAD` in the backend as revision, the implementation would default to main, if such an branch exists - therefore, it would suggest a wrong branch. 

Another example is one large model inside DB, where the branch "nightly" is used as main branch. 

In general, a specific and existing branch or tag should be used when assigning a repository. We should rather validate the branch or tag selection with a validator in the frontend.